services/server/src/handlers/topic.ts

```diff
@@ -31,6 +32,12 @@ const generalUserTopicsBodyParser = zod.object({
   topicIds: zod.array(zod.number()),
 });
 
+const replaceUserTopicsBodyParser = zod.object({
+  oldTopics: zod.array(zod.number()),
+  newTopics: zod.array(zod.number()),
```

Comment:
You can accept just the new topics. You can get all the old topics from the database. 

```suggestion
  topics: zod.array(id),
```